zst-volatile-macro/src/lib.rs

```diff
@@ -49,3 +55,35 @@ pub fn derive_volatile(tts: TokenStream) -> TokenStream {
     }
     .into()
 }
+
+fn parse_repr(attrs: &[Attribute]) -> Result<bool, &'static str> {
+    let repr = attrs
+        .iter()
+        .find(|a| a.path.is_ident("repr"))
+        .ok_or("VolatileStruct can only apply to #[repr(C, ...)]")?;
+
+    let repr_metas = match repr.parse_meta().unwrap() {
+        syn::Meta::List(l) => l,
+        _ => return Err("#[repr(...)] invalid format"),
+    };
+
+    // Make sure #[repr(C)] exits
+    repr_metas
+        .nested
+        .iter()
+        .find(|m| match m {
+            syn::NestedMeta::Meta(syn::Meta::Path(p)) => p.is_ident("C"),
+            _ => false,
+        })
+        .ok_or("VolatileStruct can only apply to #[repr(C, ...)]")?;
+
+    // Check if packed struct
+    Ok(repr_metas
+        .nested
+        .iter()
+        .find(|m| match m {
+            syn::NestedMeta::Meta(syn::Meta::Path(p)) => p.is_ident("packed"),
```

Comment:
Note that the packed attribute can also contain a value such e.g. `#[repr(packed(2))]` in which case fields aren't packed to 1 byte. Also note that the `packed` attribute can be specified more than once.

Reply:
Didn't know that before. But there is a new problem, how to encode this `packed(N)` info into alignment calculation?